apps/editing-toolkit/editing-toolkit-plugin/wpcom-global-styles/index.php

```diff
@@ -34,26 +34,42 @@ function wpcom_should_limit_global_styles( $blog_id = 0 ) {
 		}
 	}
 
-	// Do not limit Global Styles on sites created before we made it a paid feature (2022-12-15).
-	if ( $blog_id < 213403000 ) {
+	// Do not limit Global Styles on demo sites.
+	if ( wpcom_global_styles_has_blog_sticker( 'theme-demo-site', $blog_id ) ) {
 		return false;
 	}
 
-	if ( wpcom_site_has_feature( WPCOM_Features::GLOBAL_STYLES, $blog_id ) ) {
-		return false;
-	}
-
-	if ( function_exists( 'has_blog_sticker' ) && has_blog_sticker( 'theme-demo-site', $blog_id ) ) {
+	// Do not limit Global Styles on sites created before we made it a paid feature (2022-12-15),
+	// that had already used Global Styles.
+	if ( wpcom_premium_global_styles_is_site_exempt( $blog_id ) ) {
 		return false;
 	}
 
-	if ( function_exists( 'wpcomsh_is_site_sticker_active' ) && wpcomsh_is_site_sticker_active( 'theme-demo-site' ) ) {
+	// Do not limit Global Styles if the site paid for it.
+	if ( wpcom_site_has_feature( WPCOM_Features::GLOBAL_STYLES, $blog_id ) ) {
 		return false;
 	}
 
 	return true;
 }
 
+/**
+ * Wrapper to test a blog sticker on both Simple and Atomic sites at once.
+ *
+ * @param string $blog_sticker The blog sticker.
+ * @param int    $blog_id The WPCOM blog ID.
+ * @return bool Whether the site has the blog sticker.
+ */
+function wpcom_global_styles_has_blog_sticker( $blog_sticker, $blog_id ) {
+	if ( function_exists( 'has_blog_sticker' ) && has_blog_sticker( $blog_sticker, $blog_id ) ) {
+		return true;
+	}
+	if ( function_exists( 'wpcomsh_is_site_sticker_active' ) && wpcomsh_is_site_sticker_active( $blog_sticker ) ) {
```

Comment:
Since we are hardcoding the result for Atomic sites in `wpcom_premium_global_styles_is_site_exempt`, is this code necessary? 🤔

Reply:
It's necessary so that the check for `theme-demo-site` works.

It's unnecessary for anything else, only four stickers get synced to atomic sites - `woop`, `wpcom-marketplace`, `theme-demo-site` and `difm-lite-in-progress` See:
 * fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Szh%2Qcyhtvaf%2Sjcpbz%2Qoybt%2Qfgvpxref.cuc%3Se%3Qp0n1pn5r%23138-og for the list of stickers
 * fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Szh%2Qcyhtvaf%2Sjcpbz%2Qoybt%2Qfgvpxref.cuc%3Se%3Qp0n1pn5r%2331-og for `add_block_sticker`
 * https://github.com/Automattic/wpcomsh/blob/7e9d0d35d8359b5866fd68cde364171cac016a3c/feature-plugins/masterbar.php#L213 for `wpsomsh_is_site_sticker_active`